zebra-chain/src/types/amount.rs

```diff
@@ -274,6 +274,22 @@ mod test {
         Ok(())
     }
 
+    #[test]
+    fn test_add_opt_both() -> Result<()> {
+        zebra_test::init();
+        let one: Amount = 1.try_into().unwrap();
+        let one = Some(one);
+        let neg_one: Amount = (-1).try_into().unwrap();
+        let neg_one = Some(neg_one);
+
+        let zero: Amount = 0.try_into().unwrap();
+        let new_zero = one.and_then(|one| one + neg_one);
```

Comment:
Question: Is there a reason why we don't impl Add with option on both sides?

Reply:
because rust says it's a crime. Or more accurately it violates the orphan rules for trait impls which state you can't impl a foreign trait for a foreign type. In this case it's a little more complex because of the type parameter on the trait which makes covered types start to matter. for more details on the coherence rules check out https://rust-lang.github.io/rfcs/2451-re-rebalancing-coherence.html